src/components/accessibility-disability-simulator/accessibility-disability-simulator.js

```diff
@@ -0,0 +1,229 @@
+			 * @type {Number}
+			 */
+			blurLevel: { type: Number, attribute: 'blur-level', reflect: true },
+			/**
+			 * The type of disability to simulate
+			 * @type {'no-vision'|'low-vision'|'keyboard-only'|'colorblind-achromatopsia'|'colorblind-deuteranopia'|'colorblind-protanopia'|'colorblind-tritanopia'}
+			 */
+			disabilityType: { type: String, attribute: 'disability-type', reflect: true },
+			/**
+			 * Whether or not the alert should be shown
+			 * @type {Boolean}
+			 */
+			hideAlert: { type: Boolean, attribute: 'hide-alert' },
+			/**
+			 * Whether or not to show the disability type selection and blur level controls
+			 * @type {Boolean}
+			 */
+			hideControls: { type: Boolean, attribute: 'hide-controls' }
+		};
+	}
+
+	static get styles() {
+		return [ bodyCompactStyles, bodyStandardStyles, offscreenStyles, selectStyles, css`
+			.simulator-controls {
+				align-items: center;
+				display: flex;
+				gap: 0.5rem;
+				margin-bottom: 1rem;
+			}
+
+			.low-vision-slider {
+				align-items: center;
+				display: flex;
+				gap: 0.5rem;
+			}
+
+			.alert-message {
+				display: inline-block;
+				margin-bottom: 1rem;
+				padding: 0.5rem;
+			}
+
+			.svg-filter {
+				display: none;
+			}
+
+			:host([disability-type="keyboard-only"]) .wrapper-slot {
+				pointer-events: none;
+			}
+
+			:host([disability-type="no-vision"]) .wrapper-slot {
+				filter: grayscale(1);
+			}
+
+			:host([disability-type="colorblind-achromatopsia"]) .wrapper-slot {
+				filter: grayscale(1);
+			}
+
+			:host([disability-type="colorblind-deuteranopia"]) .wrapper-slot {
+				filter: url("#deuteranopia");
+			}
+
+			:host([disability-type="colorblind-protanopia"]) .wrapper-slot {
+				filter: url("#protanopia");
+			}
+
+			:host([disability-type="colorblind-tritanopia"]) .wrapper-slot {
+				filter: url("#tritanopia");
+			}
+		` ];
+	}
+
+	constructor() {
+		super();
+		this.blurLevel = 50;
+	}
+
+	render() {
+		const wrapperClasses = {
+			'wrapper-slot': true,
+			'd2l-offscreen': this.disabilityType === DISABILITY_TYPES.noVision
+		};
+		const wrapperStyles = {
+			...(this.disabilityType === DISABILITY_TYPES.lowVision && { filter: `blur(${this.blurLevel / 20}px)` })
+		};
+		return html`
+			${this._renderColorblindFilters()}
+			${this._renderControls()}
+			${this._renderAlert()}
+
+			<div class="${classMap(wrapperClasses)}" style="${styleMap(wrapperStyles)}">
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		if (changedProperties.has('blurLevel')) {
+			this.blurLevel = Math.max(Math.min(this.blurLevel, 100), 1);
+		}
+	}
+
+	_localize(key) {
+		return this.localize(`components:accessibilityDisabilitySimulator:${key}`);
+	}
+
+	_onDisabilityTypeChanged(e) {
+		this.disabilityType = e.target.value;
+	}
+
+	_onSliderChanged(e) {
+		this.blurLevel = e.target.value;
+	}
+
+	_renderAlert() {
+		if (this.hideAlert || this.disabilityType !== DISABILITY_TYPES.noVision) return;
```

Comment:
It might make this more readable if you moved the check for `disabilityType` to render before calling `_renderAlert` (e.g., `this.disabilityType === DISABILITY_TYPES.noVision ? this._renderAlert() : nothing`)

Reply:
The main reason I didn't do that is in the event that we add more alerts like this later, then it would probably be better to have those longer checks in the `_renderAlert` function instead.